src/ints/bios.cpp

```diff
@@ -946,64 +951,84 @@ static Bitu INT15_Handler(void) {
 	case 0xc2:	/* BIOS PS2 Pointing Device Support */
 		switch (reg_al) {
 		case 0x00:                      // enable/disable
-			if (reg_bh==0) {	// disable
-				MOUSE_SetPS2State(false);
-				reg_ah=0;
+			if (reg_bh == 0) {      // disable
+				MOUSEBIOS_SetState(false);
+				reg_ah = 0;
 				CALLBACK_SCF(false);
-			} else if (reg_bh==0x01) {	//enable
-				if (!MOUSE_SetPS2State(true)) {
-					reg_ah=5;
+			} else if (reg_bh == 0x01) { // enable
+				if (!MOUSEBIOS_SetState(true)) {
+					reg_ah = 5;
 					CALLBACK_SCF(true);
 					break;
 				}
-				reg_ah=0;
+				reg_ah = 0;
 				CALLBACK_SCF(false);
 			} else {
 				CALLBACK_SCF(true);
-				reg_ah=1;
+				reg_ah = 1;
 			}
 			break;
-		case 0x01:               // reset
+		case 0x01: // reset
+			MOUSEBIOS_Reset();
 			reg_bx = 0x00aa; // mouse
 			[[fallthrough]];
 		case 0x05:		// initialize
-			if ((reg_al==0x05) && (reg_bh!=0x03)) {
-				// non-standard data packet sizes not supported
+			if ((reg_al == 0x05) && !MOUSEBIOS_SetPacketSize(reg_bh)) {
 				CALLBACK_SCF(true);
-				reg_ah=2;
+				reg_ah = 2;
 				break;
 			}
-			MOUSE_SetPS2State(false);
+			MOUSEBIOS_SetState(false);
 			CALLBACK_SCF(false);
 			reg_ah=0;
 			break;
 		case 0x02:		// set sampling rate
-		case 0x03:		// set resolution
+			if (!MOUSEBIOS_SetSampleRate(reg_bh)) {
+				CALLBACK_SCF(true);
+				reg_ah = 2;
+				break;
+			}
 			CALLBACK_SCF(false);
-			reg_ah=0;
+			reg_ah = 0;
+			break;
+		case 0x03: // set resolution
+			if (!MOUSEBIOS_SetResolution(reg_bh)) {
+				CALLBACK_SCF(true);
+				reg_ah = 2;
+				break;
+			}
+			CALLBACK_SCF(false);
+			reg_ah = 0;
 			break;
-		case 0x04:		// get type
-			reg_bh=0;	// ID
+		case 0x04: // get type
+			reg_bh = MOUSEBIOS_GetType();
 			CALLBACK_SCF(false);
 			reg_ah=0;
 			break;
-		case 0x06:		// extended commands
-			if ((reg_bh==0x01) || (reg_bh==0x02)) {
-				CALLBACK_SCF(false); 
-				reg_ah=0;
+		case 0x06: // extended commands
+			if (reg_bh == 0x00) { // get mouse status
```

Comment:
Maybe introduce constants for the various commands?

Reply:
IMHO this is not a good idea - when someone digs into such file, he usually has Ralf Brown Interrupt List (or something similar) open for reference, and there are no constants there, just pure hex IDs. Introducing constants would make analysis harder - comments should be enough.

Reply:
Sounds reasonable, the comments would be better.